### PR TITLE
feat: add targetEndpoint property to ask-resources.json

### PR DESCRIPTION
**Pull request by kakhaUrigashvili**
add targetEndpoint property to ask-resources.json.

Tested with CFN and Lambda deployer. If lambda/skill already exists, the permission for new skill type (smart home for example) need to be added manually. If creating fresh, it should all work.

Test
"targetEndpoint": ["custom","smartHome", "events"]
CF Deployer
```
ask deploy
Deploy configuration loaded from ask-resources.json
Deploy project for profile [default]

==================== Deploy Skill Metadata ====================
[Warn]: The hash of current skill package folder does not change compared to the last deploy hash result, CLI will skip the deploy of skill package.
Skill ID: amzn1.ask.skill.4f5eecb6-1069-4bb4-9299-927f213c30fd

==================== Build Skill Code ====================
Skill code built successfully.
Code for region default built to /Users/kakuri/test/tend/tendcf/.ask/lambda/build.zip successfully with build flow NodeJsNpmBuildFlow.

==================== Deploy Skill Infrastructure ====================
  ✔ Deploy Alexa skill infrastructure for region "default"
Skill infrastructures deployed successfully through @ask-cli/cfn-deployer.

==================== Enable Skill ====================
[Warn]: CliWarn: Skill with multiple api domains cannot be enabled. Skip the enable process.
```

Lambda Deployer
```
ask deploy
Deploy configuration loaded from ask-resources.json
Deploy project for profile [default]

==================== Deploy Skill Metadata ====================
[Warn]: The hash of current skill package folder does not change compared to the last deploy hash result, CLI will skip the deploy of skill package.
Skill ID: amzn1.ask.skill.376ca5de-ad84-4e66-8f7e-e6d6b7401f36

==================== Build Skill Code ====================
Skill code built successfully.
Code for region default built to /Users/kakuri/test/tend/.ask/lambda/build.zip successfully with build flow NodeJsNpmBuildFlow.

==================== Deploy Skill Infrastructure ====================
  ✔ Deploy Alexa skill infrastructure for region "default"
Skill infrastructures deployed successfully through @ask-cli/lambda-deployer.

==================== Enable Skill ====================
[Warn]: CliWarn: Skill with multiple api domains cannot be enabled. Skip the enable process.
```

### CODE REVIEW

**Review by RonWang — APPROVED** [review]
Overall LGTM! 

Approve in advance, please just update the Markdown file before push.

File: lib/controllers/skill-infrastructure-controller/index.js [hunk 1]
```diff
@@ -140,12 +140,20 @@ module.exports = class SkillInfrastructureController {
      * @param {Function} callback (error)
      */
     updateSkillManifestWithDeployResult(rawDeployResult, callback) {
-        // 1.update local skill.json file: update the "uri" in all the existing "apis" for each region
-        R.keys(Manifest.getInstance().getApis()).forEach((domain) => {
+        const targetDomains = ResourcesConfig.getInstance().getTargetDomains(this.profile);
+        // for backward compatibility, defaulting to api from skill manifest if targetEndpoints is not defined
+        const domains = targetDomains.length ? targetDomains : Object.keys(Manifest.getInstance().getApis());
```
Comment (kakhaUrigashvili) [inline lib/controllers/skill-infrastructure-controller/index.js:145]: duplicate logic. same logic in lambda deployer. not sure what is the best place to put

File: lib/controllers/skill-infrastructure-controller/index.js [hunk 2]
```diff
@@ -140,12 +140,20 @@ module.exports = class SkillInfrastructureController {
...
+        // 1.update local skill.json file: update the "uri" in all target endpoints for each region
+        domains.forEach((domain) => {
             R.keys(rawDeployResult).forEach((region) => {
                 Manifest.getInstance().setApisEndpointByDomainRegion(domain, region, rawDeployResult[region].endpoint);
             });
         });
+
+        if (ResourcesConfig.getInstance().hasEventsEndpoint(this.profile)) {
+            Manifest.getInstance().setEventEndpoint(rawDeployResult.default.endpoint.uri);
```
Comment (RonWang) [inline lib/controllers/skill-infrastructure-controller/index.js:154]: what if the endpoint is not for default? the "events" object also has default endpoint uri and "regions" endpoint uri.

Comment (kakhaUrigashvili) [reply lib/controllers/skill-infrastructure-controller/index.js]: updated

**Review by RonWang — APPROVED** [review]
Ship it!

**Pull request merged**